src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -548,7 +547,29 @@ internal List<INodeLibraryDependencyInfo> NodeLibraryDependencies
                             packageDependencies[saved] = new PackageDependencyInfo(saved);
                         }
                         packageDependencies[saved].AddDependent(node.GUID);
-                        packageDependencies[saved].IsLoaded = saved.Equals(collected);
+
+                        if (collected != null)
+                        {
+                            if (saved.Name == collected.Name)
+                            {
+                                if (saved.Version == collected.Version)
+                                {
+                                    packageDependencies[saved].State = PackageDependencyState.Loaded;
```

Comment:
One thing that has seemed weird to me here (since even before this PR) is that package state gets set “per node” rather than “per package”. So in the course of this For loop, a package’s state gets set not once, but once per node dependent on the package. When I originally wrote this, I decided this wasn’t really an issue because all of the dependent nodes would agree on the state—it was either “loaded” or “not loaded”. Is this assumption still fair? What about a case where some (but not all) nodes in a missing package are resolved by another package?

Reply:
Good point Scott, we will have to discuss these cases.